main.tf

```diff
@@ -220,6 +221,21 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = !(var.microsoft_defender_enabled && !var.log_analytics_workspace_enabled)
       error_message = "Enabling Microsoft Defender requires that `log_analytics_workspace_enabled` be set to true."
     }
+    precondition {
```

Comment:
Since the `condition` for this `precondition` is complex, I would rather to create a new `local` ,in `locals.tf` file, and add some unit tests for this condition check. Would you please refer the current unit test inside `test/unit` folder and help us by adding three unit test cases to cover this `precondition`? Thanks!

Reply:
Good point. I'll move it into `locals.tf` and add a unit test.

Reply:
>I though about your suggestion of putting the check into locals.tf but can locals contain a precondition check?

Could we extract the expression into a local varialbe with `bool` type? Like:

```hcl
locals{
automatic_channel_upgrade_check = (var.automatic_channel_upgrade == null || (
contains(["patch"], var.automatic_channel_upgrade) &&
        regex("^[0-9]{1,}\\.[0-9]{1,}$", var.kubernetes_version)
        ) || (
        # or we have set it to rapid/stable and don't specify the version at all
        contains(["rapid", "stable", "node-image"], var.automatic_channel_upgrade) &&
        var.kubernetes_version == null
        )
      )
}
```

Then the `precondition` could be:

```hcl
precondition {
      // either we have automatic upgrades disabled
      condition = local.automatic_channel_upgrade_check
...
```

Reply:
I've moved the condition into a bool var.